### ISSUE TITLE
Struts2 i18n is not working with mustache.js

### ISSUE THREAD

**Issue by sathishixm** [janl/mustache.js#638]
In my application we are using Struts2 framework. After seeing the benefits of mustache.js, I have integrated mutache template in my application.

Here is the flow, 
1. I am loading values from JSON through ajax. Here is the JSON
   {
	"informationType": [{
		"infotype": "ANCA",
		"subtype": ["AS", "AI", "AG", "BS", "BB"]
	}, {
		"infotype": "HW",
		"subtype": ["SM", "RPQ"]
	}]
   }
2. I am rendering the loaded JSON through mustache.js templating.
   
    this.$select.html(Mustache.render(this.template,this._informationType));

3. Here is the template from my jsp page.
    <select id="infoOptions" multiple style="width: 380px;" data-placeholder="Information Type">
                        <script id="infotype-template" type=text/template>
                            {{#informationType}}
                            {{#infotype}}
                            <option class="optionGroup" value="{{.}}">{{.}}</option>
                            {{/infotype}}
                                {{#subtype}}
                                <option  class="optionChild" value="{{.}}">{{.}}</option>
                                {{/subtype}}
                            {{/informationType}}
                        </script>
                        </select>

It works perfectly and it loads the values exactly as in my json. However, what I need is to load the translated text based on my json value though Struts2 i18n.

So I tried passing the {{.}} to <s:text name="{{.}}"/> its not picking the translated values rather printing the same text from my JSON.

Here is the code I have tried,

<select id="infoOptions" multiple style="width: 380px;" data-placeholder="Information Type">
                        <script id="infotype-template" type=x-tmpl-mustache>
                            {{#informationType}}
                            {{#infotype}}
                            <option class="optionGroup" value="<s:text name="{{.}}"/>">**<s:text name="{{.}}"/>**</option>
                            {{/infotype}}	
                                {{#subtype}}
                                <option  class="optionChild" value="<s:text name="{{.}}"/>"><s:text name="{{.}}"/></option>
                                {{/subtype}}
                            {{/informationType}}
                        </script>
                        </select>

I assume that the {{.}} returns the object type so its not loading the value from value stack. 
Note: If i pass the hardcoded json value like this "<s:text name="ANCA"/> rather <s:text name="{{.}}"/> its returning values. 

can anyone help solve this problem. Thanks in advance.

**Comment by phillipj** [comment]
Just to break the silence here, I have never worked with Struts so I'm not sure I'll be much of help to you, but I know not getting any feedback are more frustrating.

Have you implemented this with server side rendering w/mustache.java or is it only client side w/mustache.js so far? If the latter, I'm suspecting the `<s:text name="{{.}}"/>` is something executed by the server, before sent to the client? If so the server/Struts possibly doesn't know what `{{.}}` means and therefore doesn't behave as you'd expect.

A possible solution without having to implement mustache on the server as well, is translating the strings **before** putting them into the JSON payload sent the client.

**Comment by sathishixm** [comment]
Thanks a lot for taking your time to answer my question.... Yes, I have used Struts ognl tags  to populate all the values from json to front end, which provides the same advantage of using mutache... Thanks again. :)
